### PR TITLE
Throughput shell interface

### PR DESCRIPTION
**Pull request by KAGA164**
This PR added a shell interface to the Throughput sample. The Throughput parameters can be changed every time before running a test

### CODE REVIEW

**Comment by b-gent** [comment]
Assigning doc review to @peknis

File: samples/bluetooth/throughput/README.rst [hunk 1]
```diff
@@ -10,12 +10,12 @@ Bluetooth: Throughput
...
 You can use it to determine the maximum throughput, or to experiment with different connection parameters and check their influence on the throughput.
 
-
 Overview
 ********
 
 The sample transmits data between two boards, the *tester* and the *peer*, and measures the throughput performance.
-To do so, it uses the :ref:`throughput_readme`.
+To do so, it uses the :ref:`throughput_readme`. You can run the tests by connecting to the board through the serial port and sending shell commands.
+Zephyr's :ref:`zephyr:shell_api` module is used to handle the commands
```
Comment (peknis) [inline samples/bluetooth/throughput/README.rst:18]: Proposing:
It uses the ...
To run the tests, connect to the board using the serial port and send shell commands.

File: samples/bluetooth/throughput/README.rst [hunk 2]
```diff
@@ -55,15 +55,21 @@ By default, the following connection parameter values are used:
 Changing connection parameter values
 ====================================
 
-You can experiment with different connection parameter values by reconfiguring the values using menuconfig and then compiling and programming the sample again to at least one of the boards.
+You can experiment with different connection parameter values by reconfiguring the values using the :ref:`zephyr:shell_api` interface before running a test.
```
Comment (peknis) [inline samples/bluetooth/throughput/README.rst:58]: To experiment with different connection parameter values, reconfigure the values...

File: samples/bluetooth/throughput/README.rst [hunk 3]
```diff
@@ -55,15 +55,21 @@ By default, the following connection parameter values are used:
...
+The following parameters can be adjusted:
```
Comment (peknis) [inline samples/bluetooth/throughput/README.rst:59]: You can adjust the following parameters:

File: samples/bluetooth/throughput/README.rst [hunk 4]
```diff
@@ -55,15 +55,21 @@ By default, the following connection parameter values are used:
...
    In a *Bluetooth* Low Energy connection, the different devices negotiate the connection parameters that are used.
    If the configuration parameters for the devices differ, they agree on the lowest common denominator.
 
    By default, the sample uses the fastest connection parameters.
-   If you change them to lower values, it is sufficient to program them to one of the boards.
-   If you were to change them to higher values, you would need to program both boards again.
+   You can change them to different valid values without a need to program the boards again.
+
+.. note::
+   When LE Connection Interval is set to high values and the PHY or the Data Length needs to be changed in the next test then the PHY Update procedure or Data Length Update procedure can take several seconds.
```
Comment (peknis) [inline samples/bluetooth/throughput/README.rst:72]: When you have set the LE Connection Interval to high values and need to change the PHY or the Data Length in the next test, the PHY...

File: samples/bluetooth/throughput/README.rst [hunk 5]
```diff
@@ -89,6 +95,16 @@ You can use any two of the development kits listed above and mix different devel
 
 The sample also requires a connection to a computer with a serial terminal for each of the development kits.
 
+User interface
+**************
+
+Button 1:
+   Put the board into a master (tester) role.
```
Comment (peknis) [inline samples/bluetooth/throughput/README.rst:102]: Put -> Set

File: samples/bluetooth/throughput/README.rst [hunk 6]
```diff
@@ -104,16 +120,21 @@ After programming the sample to both boards, test it by performing the following
 1. Connect to both boards with a terminal emulator (for example, PuTTY).
    See :ref:`putty` for the required settings.
 #. Reset both boards.
-#. In one of the terminal emulators, type "s" to start the application on the connected board in the slave (peer) role.
-#. In the other terminal emulator, type "m" to start the application in the master (tester) role.
+#. Press Button 1 on the board to put the board into master (tester) role.
```
Comment (peknis) [inline samples/bluetooth/throughput/README.rst:123]: put -> set

File: samples/bluetooth/throughput/README.rst [hunk 7]
```diff
@@ -104,16 +120,21 @@ After programming the sample to both boards, test it by performing the following
...
        Ready, press any key to start
 
-#. Press a key in the terminal that is connected to the tester.
+#. Type ``config print`` in the terminal to print the current configuration.
+   Type ``config`` in the terminal to configure the test parameters to your choice.
+   Use the Tab key for auto-completion and to view the options available for a parameter.
+#. Type ``run`` in the terminal to start the test.
 #. Observe the output while the tester sends data to the peer.
    At the end of the test, both tester and peer display the results of the test.
+#. Repeat the test after having changed the parameters.
```
Comment (peknis) [inline samples/bluetooth/throughput/README.rst:136]: ... after changing the parameters.

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/bluetooth/throughput/README.rst (hunk 1 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/README.rst:19]: Done

File: samples/bluetooth/throughput/README.rst (hunk 2 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/README.rst]: Done

File: samples/bluetooth/throughput/README.rst (hunk 3 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/README.rst]: Done

File: samples/bluetooth/throughput/README.rst (hunk 4 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/README.rst]: Done

File: samples/bluetooth/throughput/README.rst (hunk 5 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/README.rst]: Fixed

File: samples/bluetooth/throughput/README.rst (hunk 6 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/README.rst]: Fixed

File: samples/bluetooth/throughput/README.rst (hunk 7 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/README.rst]: Done

**Review by peknis — APPROVED** [review]

**Review by grochu — COMMENTED** [review]
Looks good overall, I left a few minor comments.

File: samples/bluetooth/throughput/prj_nrf5340pdk_nrf5340_cpuapp.conf [hunk 8]
```diff
@@ -6,7 +6,7 @@
 
 CONFIG_CONSOLE_SUBSYS=y
 CONFIG_CONSOLE_HANDLER=y
```
Comment (lemrey) [inline samples/bluetooth/throughput/prj_nrf5340pdk_nrf5340_cpuapp.conf:8]: Do we still need these?

File: samples/bluetooth/throughput/src/cmds.c [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+
+#if defined(RADIO_MODE_MODE_Ble_LR500Kbit) || defined(NRF5340_XXAA_APPLICATION)
+static int cmd_phy_coded_s8(const struct shell *shell, size_t argc,
+			    char **argv)
+{
+	test_params.phy->options = BT_CONN_LE_PHY_OPT_CODED_S8;
+	test_params.phy->pref_rx_phy = BT_GAP_LE_PHY_CODED;
+	test_params.phy->pref_tx_phy = BT_GAP_LE_PHY_CODED;
+
+	shell_print(shell, "Preferred PHY set to: %s",
```
Comment (lemrey) [inline samples/bluetooth/throughput/src/cmds.c:132]: nit: `PHY set to: ...` for consistency with other prints

**Review by lemrey — APPROVED** [review]
Nice :)
nit: please update the commit message `samples: bluetooth: throughput: ...`

**Comment by joerchan** [comment]
@qung @KAGA164 Is this sample working in CI with these changes now? Input of roles has switched from UART input, to buttons.

File: samples/bluetooth/throughput/src/cmds.c [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+	shell_print(shell, "PHY set to: %s", phy_str(test_params.phy));
+
+	return 0;
+}
+
+#if defined(RADIO_MODE_MODE_Ble_LR125Kbit) || defined(NRF5340_XXAA_APPLICATION)
+static int cmd_phy_coded_s2(const struct shell *shell, size_t argc,
+			    char **argv)
+{
+	test_params.phy->options = BT_CONN_LE_PHY_OPT_CODED_S2;
```
[the remark this answers is not in the document]
Comment (joerchan) [reply samples/bluetooth/throughput/src/cmds.c:112]: Correct, LR125kbit is S8.

File: samples/bluetooth/throughput/prj_nrf5340pdk_nrf5340_cpuapp.conf (hunk 8 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/prj_nrf5340pdk_nrf5340_cpuapp.conf]: We don't ;) Removed

File: samples/bluetooth/throughput/src/cmds.c (hunk 9 above)
Comment (KAGA164) [reply samples/bluetooth/throughput/src/cmds.c]: Ok. Changed

File: samples/bluetooth/throughput/src/cmds.c [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+#define MIN_CONN_INTERVAL   6
+#define MAX_CONN_INTERVAL   3200
+#define SUPERVISION_TIMEOUT 1000
+
+static struct test_params {
+	struct bt_le_conn_param *conn_param;
+	struct bt_conn_le_phy_param *phy;
+	struct bt_conn_le_data_len_param *data_len;
+} test_params = {
+	.conn_param = BT_LE_CONN_PARAM(0x140, 0x140, 0, 400),
```
[the remark this answers is not in the document]
Comment (KAGA164) [reply samples/bluetooth/throughput/src/cmds.c]: Yes sure. Done

File: samples/bluetooth/throughput/src/cmds.c (hunk 10 above)
[the remark this answers is not in the document]
Comment (KAGA164) [reply samples/bluetooth/throughput/src/cmds.c:117]: Yes, you are right. The S8 phy is equal to LR125kbit. Fixed

File: samples/bluetooth/throughput/src/cmds.c [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+		    test_params.conn_param->interval_min,
+		    phy_str(test_params.phy));
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(phy_sub,
+	SHELL_CMD(1M, NULL, "Set preferred PHY to 1Mbps", cmd_phy_1m),
+	SHELL_CMD(2M, NULL, "Set preferred PHY to 2Mbps", cmd_phy_2m),
+#if defined(RADIO_MODE_MODE_Ble_LR125Kbit) || defined(NRF5340_XXAA_APPLICATION)
+	SHELL_CMD(coded_s2, NULL, "Set preferred PHY to Coded S2",
```
[the remark this answers is not in the document]
Comment (KAGA164) [reply samples/bluetooth/throughput/src/cmds.c:236]: Fixed

**Review by qung — APPROVED** [review]

**Comment by KAGA164** [comment]
@joerchan The CI must be changed for this sample. @qung prepared a needed changes in the CI.

**Review by grochu — APPROVED** [review]

**Comment by carlescufi** [comment]
@KAGA164 can this be merged now?

**Comment by KAGA164** [comment]
@carlescufi This PR can be merged now.

**Pull request merged**